### PR TITLE
Unhook methods and test fix

### PR DESCRIPTION
**Pull request by franzigeiger**
- Wrong import because of recent changes
- Unhook methods

### CODE REVIEW

File: model_tools/activations/pca.py [hunk 1]
```diff
@@ -75,7 +75,10 @@ def init_and_progress(layer, activations):
     @classmethod
     def hook(cls, activations_extractor, n_components):
         hook = LayerPCA(activations_extractor=activations_extractor, n_components=n_components)
-        assert not cls.is_hooked(activations_extractor), "PCA already hooked"
+        hooks = activations_extractor._extractor._batch_activations_hooks
```
Comment (mschrimpf) [inline model_tools/activations/pca.py]: this access two protected attributes which seems prone to maintenance issues in the future. If we leave it like this, we definitely need unit tests ensuring this functionality.

File: model_tools/activations/pca.py [hunk 2]
```diff
@@ -75,7 +75,10 @@ def init_and_progress(layer, activations):
...
+        for key in list(hooks):
+            if isinstance(hooks[key], cls):
+                del hooks[key]
```
Comment (mschrimpf) [inline model_tools/activations/pca.py]: Isn't this case indicative of a bug? Someone has already hooked PCA and is hooking it again? I would raise an Error here.

File: model_tools/activations/pca.py (hunk 1 above)
Comment (franzigeiger) [reply model_tools/activations/pca.py]: Hm I could change it to a function call of this activation extractor. And as you may have seen there is a unittest for this functionality in candidate_models. If you think it's important I can add one here as well.

Comment (mschrimpf) [reply model_tools/activations/pca.py]: Yes I think just adding a unit test in this repository would suffice for now

Comment (franzigeiger) [reply model_tools/activations/pca.py]: The change is reverted. I found a more elegant way.

File: model_tools/activations/pca.py (hunk 2 above)
Comment (franzigeiger) [reply model_tools/activations/pca.py]: Also reverted

**Comment by mschrimpf** [comment]
Seems like the last commit undid all the changes, is this PR out-dated?

**Comment by franzigeiger** [comment]
No it's not. I found a better way to solve the hook problem, so this ugly unhooking isn't necessary any more.

**Pull request merged**